backend/dal/requestDAL.go

```diff
@@ -19,6 +21,16 @@ func NewRequestDAL(db *sqlx.DB) *RequestDAL {
 	}
 }
 
+// Gets the database connection to be used in requests
+func getDB() *sqlx.DB {
```

Comment:
You don't need to get the db in the DAL. A db connection is created once in the main and passed to the DAL through the controller. That's why in the test there's a getDB function that is passed to the DAL when it's created.

Reply:
Removed function, it shouldn't be there. It wasn't even being used.